crates/typst/src/layout/grid/layout.rs

```diff
@@ -322,11 +369,30 @@ impl CellGrid {
                 );
             }
 
-            *slot = Some(cell);
+            *slot = Some(Entry::Cell(cell));
+
+            // Now, if the cell spans more than one column, we fill the spanned
+            // positions in the grid with Entry::Merged pointing to the
+            // original cell as its parent.
+            for (offset, slot) in resolved_cells[resolved_index..][..colspan]
```

Comment:
Haha, I should have linked to where I got this from: https://internals.rust-lang.org/t/pre-rfc-range-extension-syntax-and-offset-method-draft/19492

I actually used that syntax at first, but rustfmt required me to move the index to the next line, that is, to write

```rs
for (offset, slot) in resolved_cells
       [resolved_index..resolved_index + colspan]
       .something()
       .etc()
{
  // ...
}
```

And I hated that at first, so I changed it :joy:

I can keep the syntax you propose though if that'd be preferrable. :wink:

Reply:
> From the thread:
> 
> > Note, also, that I suggest always using a[i..][..n] instead of a[i..(i+n)], because it avoids dealing with the arithmetic overflow in the addition.
> 
> I am convinced!

Note that it seems to be the same as doing a `checked_add` (cf. https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=d746752122d81561198d8e27a694ca06). Either way we aren't getting a panic due to the bounds check before it, so should be fine.

Reply:
Well, I think I'll be keeping it like this for now then. I guess we'll have to modify this loop either way when rowspans are implemented, so we can take another look at it then. (Feel free to unresolve if there are additional questions.)